packages/core/src/routes/session/session.ts

```diff
@@ -46,12 +46,22 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
       }),
     }),
     async (ctx, next) => {
-      await provider.interactionDetails(ctx.req, ctx.res);
+      const {
+        params: { client_id },
+      } = await provider.interactionDetails(ctx.req, ctx.res);
       const { username, password } = ctx.guard.body;
       const type = 'SignInUsernamePassword';
       ctx.log(type, { username });
 
-      const { id } = await findUserByUsernameAndPassword(username, password);
+      const { id, roleNames } = await findUserByUsernameAndPassword(username, password);
+
+      if (String(client_id) === adminConsoleApplicationId) {
+        assertThat(
+          roleNames.includes(UserRole.Admin),
+          new RequestError({ code: 'auth.forbidden', status: 403 })
+        );
+      }
```

Comment:
should mention it's a temp solution since we only have username-and-password sign-in for AC now

Reply:
updated